### ISSUE TITLE
Installation finishes with a warning and installspec gives an error

### ISSUE THREAD

**Issue by pmdbio** [IRkernel/IRkernel#486]
I tried to install IRkernel through R (not in RStudio and with `sudo`) with this command:
`install.packages(c('repr', 'IRdisplay', 'evaluate', 'crayon', 'pbdZMQ', 'devtools', 'uuid', 'digest'))`
`devtools::install_github('IRkernel/IRkernel')`

Then it finished the installation but gave me this warning:
`Warning message:
GitHub repo contains submodules, may not function as expected!`

After that I tried:
`IRkernel::installspec(user = FALSE)`

And I got this error:
`Error in IRkernel::installspec(user = FALSE) : 
  jupyter-client has to be installed but “jupyter kernelspec --version” exited w
ith code 127.`

What should I do?

**Comment by merl-dev** [comment]
similar issue:

```
devtools::install_github('IRkernel/IRkernel')
Downloading GitHub repo IRkernel/IRkernel@master
from URL https://api.github.com/repos/IRkernel/IRkernel/zipball/master
Installing IRkernel
trying URL 'https://cran.rstudio.com/src/contrib/digest_0.6.12.tar.gz'
Content type 'application/x-gzip' length 120337 bytes (117 KB)
==================================================
downloaded 117 KB

sh: /bin/gtar: No such file or directory
sh: /bin/gtar: No such file or directory
Error in system(cmd, intern = TRUE) : error in running command
In addition: Warning messages:
1: GitHub repo contains submodules, may not function as expected! 
2: In utils::untar(src, exdir = target, compressed = "gzip") :
  ‘/bin/gtar -xf '/tmp/RtmpQLRUws/digest_0.6.12.tar.gz' -C '/tmp/RtmpQLRUws/devtools56c3459fa8e'’ returned error code 127
> 

```

**Comment by flying-sheep** [comment]
the submodule warning is irrelevant, you can ignore it.

@pmdbio if you can’t successfully call `system('jupyter kernelspec --version')` from R, there’s nothing we can do. you have to make sure that both `jupyter` and `jupyter-kernelspec` are in your $PATH environment variable. please read the warning box [here](https://irkernel.github.io/installation/)

@merl-dev it’s not a related issue. and it’s not a problem with IRkernel but with R itself, as `utils::untar` doesn’t work. please file an issue there

**Comment by pmdbio** [comment]
@flying-sheep  I checked and I have both `jupyter` and `jupyter-kernelspec` in my $PATH environment variable. Something interesting happened... When I tried to run `system('jupyter kernelspec --version)` within an R session initiated with `sudo` I got that error above. However, when I tried to run `system('jupyter kernelspec --version)` within an R session not initiated with `sudo` it showed me the version from `jupyter kernelspec`. However I still cannot run `IRkernel::installspec()` or `IRkernel::installspec(user=FALSE)` without getting an error. When I ran R without `sudo` and tried both commands for `installspec` it said that I had no permissions to `/user/local/share/jupyter` and asked me to run with `sudo` and try again. Do not know what to do.

**Comment by flying-sheep** [comment]
the PATH should not be changed with sudo, but you can try `sudo -E R`

**Comment by takluyver** [comment]
If you specify `user=TRUE`, does it still give you the same error? That should install to a different location where you can write without sudo.

**Comment by pmdbio** [comment]
@flying-sheep  I was able to run `IRkernel::installspec()` with no errors running R without `sudo` (don't know why it worked now). However, `IRkernel::installspec(user=FALSE)` still gives me the same error with or without running R with `sudo`. I tried your suggestion `sudo -E R` and still gives me the same error for `IRkernel::installspec(user=FALSE)`.

@takluyver  I think my answer to @flying-sheep  answers your question since the default value is `user = TRUE`.
